core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/BillingApiServiceV2Spec.scala

```diff
@@ -538,7 +554,51 @@ class BillingApiServiceV2Spec extends ApiServiceSpec with MockitoSugar {
       }
   }
 
-  "GET /billing/v2/{projectName}/spendReport" should "400 when invalid dates are given" in withEmptyDatabaseAndApiServices { services =>
+  "GET /billing/v2/{projectName}/spendReport" should "200 and return only summary information when aggregation key is not present" in withEmptyTestDatabase { dataSource: SlickDataSource =>
+    val mockSpendReportingService = mock[SpendReportingService](RETURNS_SMART_NULLS)
+    when(mockSpendReportingService.getSpendForBillingProject(any[RawlsBillingProjectName], any[DateTime], any[DateTime], ArgumentMatchers.eq(Set.empty)))
+      .thenReturn(Future.successful(SpendReportingResults(Seq.empty, SpendReportingForDateRange("0.0", "0.0", "USD", Option(DateTime.now()), Option(DateTime.now())))))
+
+    withApiServicesAndCustomSpendReporting(dataSource, mockSpendReportingService) { services =>
+      val project = createProject("project")
+
+      Get(s"/billing/v2/${project.projectName.value}/spendReport?startDate=2022-03-06&endDate=2022-03-07") ~>
+        sealRoute(services.billingRoutesV2) ~>
+        check {
+          assertResult(StatusCodes.OK, responseAs[String]) {
+            status
+          }
+
+          responseAs[SpendReportingResults].spendDetails shouldBe empty
+        }
+    }
+  }
+
+  it should "200 and parse aggregation keys" in withEmptyTestDatabase { dataSource: SlickDataSource =>
+    val mockSpendReportingService = mock[SpendReportingService](RETURNS_SMART_NULLS)
+    val expectedAggregationKeys = Set(
+      SpendReportingAggregationKeyWithSub(SpendReportingAggregationKeys.Workspace, Option(SpendReportingAggregationKeys.Daily)),
+      SpendReportingAggregationKeyWithSub(SpendReportingAggregationKeys.Category),
+      SpendReportingAggregationKeyWithSub(SpendReportingAggregationKeys.Category, Option(SpendReportingAggregationKeys.Category))
+    )
+    // we don't care about what it actually returns in this test, just that the mocked service is called correctly
+    when(mockSpendReportingService.getSpendForBillingProject(any[RawlsBillingProjectName], any[DateTime], any[DateTime], ArgumentMatchers.eq(expectedAggregationKeys)))
+      .thenReturn(Future.successful(SpendReportingResults(Seq.empty, SpendReportingForDateRange("0.0", "0.0", "USD", Option(DateTime.now()), Option(DateTime.now())))))
+
+    withApiServicesAndCustomSpendReporting(dataSource, mockSpendReportingService) { services =>
+      val project = createProject("project")
+
+      Get(s"/billing/v2/${project.projectName.value}/spendReport?startDate=2022-03-06&endDate=2022-03-07&aggregationKey=Workspace~Daily&aggregationKey=Category&aggregationKey=Category~Category") ~>
+        sealRoute(services.billingRoutesV2) ~>
+        check {
+          assertResult(StatusCodes.OK, responseAs[String]) {
+            status
+          }
+        }
```

Comment:
Is it possible to assert that spendDetails are present (the opposite of line 572)? Or is that pointless because it's just a mock anyway?

Reply:
I think because they're both mocked, the `spendDetails` assertion isn't worth it here or even really as valuable as it looks on 572. I'm changing both tests a little to throw if `getSpendForBillingProject` isn't called correctly and then to also verify that the method was called exactly once which covers the behavior that we care most about IMO

Reply:
Sounds good!